server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CompactionDriver.java

```diff
@@ -267,8 +267,13 @@ public int updateAndCheckTablets(Manager manager, FateId fateId)
             otherSelected++;
           }
         } else {
-          // ELASTICITY_TODO if there are compactions preventing selection of files, then add
+          // If there are compactions preventing selection of files, then add
           // selecting marker that prevents new compactions from starting
+          var mutator = tabletsMutator.mutateTablet(tablet.getExtent()).requireAbsentOperation()
+              .requireSame(tablet, ECOMP).setCompactionRequested();
```

Comment:
Its possible that two different fate ops could concurrently set this on tablet.  W/o something to make the key unique one fate op could delete what another set.  One way to avoid this problem is to make the key in the tablet metadata include the fateid, making each fate op have its own key.

```suggestion
              .requireSame(tablet, ECOMP).setCompactionRequested(fateId);
```

Reply:
I initially was going to do this but did not because when I was looking at how the compaction fate operations work I noticed that CompactRange first [reserves](https://github.com/apache/accumulo/blob/da7b3b005e13276c202b4f337496f9ecf2c039e5/server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CompactRange.java#L77) the table with a table lock before continuing to CompactionDriver so I was thinking that would prevent to concurrent fate ops from trying to set it at the same time. How would two different fate ops on the same tablet be able to set this if the table lock is needed to continue and run the CompactionDriver fate op?

Reply:
> How would two different fate ops on the same tablet be able to set this if the table lock is needed to continue and run the CompactionDriver fate op?

That line mentioned is reserving a "read lock", there is a boolean argument that determines this.  That boolean would be more clear if it were switched to an enum.   The "read lock" allows multiple concurrent compactions to grab the lock as long nothnig holds the write lock (things like create table, delete table, merge grab the write lock).

Reply:
opened #4276

Reply:
This has been changed in the latest update